From 7d261abf2593be7d62ebfec7a4b0aa1e614403e2 Mon Sep 17 00:00:00 2001
From: Rodrigo Osorio <rodrigo@FreeBSD.org>
Date: Fri, 22 Mar 2024 20:54:34 +0100
Subject: [PATCH] error: 'linux/fb.h' file not found    29 | #include
 <linux/fb.h>       |          ^~~~~~~~~~~~ error: use of undeclared
 identifier 'FBIOGET_VSCREENINFO'  1254 |         if (ioctl(fd,
 FBIOGET_VSCREENINFO, &vsinfo) < 0)       |                       ^

---
 .../codec/ddi/dec/ddi_decode_functions.cpp    | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/media_softlet/linux/common/codec/ddi/dec/ddi_decode_functions.cpp b/media_softlet/linux/common/codec/ddi/dec/ddi_decode_functions.cpp
index 4455111a7a..1d9be4b225 100644
--- a/media_softlet/linux/common/codec/ddi/dec/ddi_decode_functions.cpp
+++ b/media_softlet/linux/common/codec/ddi/dec/ddi_decode_functions.cpp
@@ -26,7 +26,39 @@
 
 #include <sys/ioctl.h>
 #include <fcntl.h>
+#if defined(__linux__)
 #include <linux/fb.h>
+#elif defined(__DragonFly__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__sun)
+#include <sys/fbio.h>
+# if defined(__sun)
+#define DEFAULT_FBDEV "/dev/fb"
+# else
+#define DEFAULT_FBDEV "/dev/ttyv0"
+# endif
+#define FBIOGET_VSCREENINFO FBIOGTYPE
+#define fb_var_screeninfo fbtype
+#define xres fb_width
+#define yres fb_height
+#elif defined(__NetBSD__) || defined(__OpenBSD__)
+#include <dev/wscons/wsconsio.h>
+# if defined(__OpenBSD__)
+#define DEFAULT_FBDEV "/dev/ttyC0"
+# else
+#define DEFAULT_FBDEV "/dev/ttyE0"
+# endif
+#define FBIOGET_VSCREENINFO WSDISPLAYIO_GINFO
+#define fb_var_screeninfo wsdisplay_fbinfo
+#define xres width
+#define yres height
+#else
+#warning "Cannot query framebuffer properties on this platform."
+#define DEFAULT_FBDEV "/dev/fb0"
+#define FBIOGET_VSCREENINFO 0
+struct fb_var_screeninfo {
+    uint32_t xres;
+    uint32_t yres;
+};
+#endif
 
 #include "ddi_decode_functions.h"
 #include "media_libva_util_next.h"
